### ISSUE TITLE
Godror attempts to query on preliminary connections

### ISSUE THREAD

**Issue by kurt-google** [godror/godror#126]
**Describe the bug**
Godror attempts to run connection initialization code for the timezones on preliminary connections where such queries will never succeed. I don't think it provokes any bad behavior as goracle also did this, but it does add extra noise to the logs.

**To Reproduce**
Attempt to startup a database on a preliminary connection with the godror package level logger set.

**Expected behavior**
Logs without errors when successfully starting the database.

**Error output**
```
msg="connect with default params" poolParams="poolIncrement=1 poolMaxSessions=1000 poolMinSessions=1 poolSessionMaxLifetime=1h0m0s\npoolSessionTimeout=5m0s poolWaitTimeout=30s" connParams="sysdba=1" common="user= password= connectString= timezone=local"
msg=dpiContext_createWithParams params="unsupported value type"
msg=acquireConn pool=null connParams="user= password= connectString= timezone=local sysdba=1"
msg="init connection" conn="&user= password= connectString= configDir= connectionClass= enableEvents=0\nheterogeneousPool=0 libDir= poolIncrement=0 poolMaxSessions=0 poolMinSessions=0\npoolSessionMaxLifetime=0s poolSessionTimeout=0s poolWaitTimeout=0s prelim=1\nstandaloneConnection=0 sysasm=0 sysdba=1 sysoper=0 timezone=local&serverVersion=0.0.0.0.0&tzOffSecs=0&new=true" onInit="unsupported value type"
msg=initTZ tzValid=false paramsTZ=UTC
msg=initTZ key="Local\tuser= password= connectString= configDir= connectionClass= enableEvents=0\nheterogeneousPool=0 libDir= poolIncrement=0 poolMaxSessions=0 poolMinSessions=0\npoolSessionMaxLifetime=0s poolSessionTimeout=0s poolWaitTimeout=0s prelim=1\nstandaloneConnection=0 sysasm=0 sysdba=1 sysoper=0 timezone=local"
enter=bindVars st=0xc00062f080 args="unsupported value type"
doManyCount=1 arrLen=-1 doExecMany=false minArrLen=maxArrLen
msg=statement.closeNotLocking st=0xc00062f080 refCount=1
msg=maybeBadConn err="dpiStmt_execute: ORA-01012: not logged on\nProcess ID: 0\nSession ID: 0 Serial number: 0" errS="\"dpiStmt_execute: ORA-01012: not logged on\\nProcess ID: 0\\nSession ID: 0 Serial number: 0\"" errT=false errV="&fmt.wrapError{msg:\"dpiStmt_execute: ORA-01012: not logged on\\nProcess ID: 0\\nSession ID: 0 Serial number: 0\", err:(*godror.OraErr)(0xc0006161e0)}"
msg="maybeBadConn close" conn="&user= password= connectString= configDir= connectionClass= enableEvents=0\nheterogeneousPool=0 libDir= poolIncrement=0 poolMaxSessions=0 poolMinSessions=0\npoolSessionMaxLifetime=0s poolSessionTimeout=0s poolWaitTimeout=0s prelim=1\nstandaloneConnection=0 sysasm=0 sysdba=1 sysoper=0 timezone=local&serverVersion=0.0.0.0.0&tzOffSecs=0&new=true" error="dpiStmt_execute: ORA-01012: not logged on\nProcess ID: 0\nSession ID: 0 Serial number: 0"
qry="SELECT DBTIMEZONE as dbTZ, NVL(TO_CHAR(SYSTIMESTAMP, 'TZR'), TO_CHAR(SYSTIMESTAMP, 'TZH:TZM')) AS dbOSTZ FROM DUAL" error="driver: bad connection"
msg=PrepareContext shortcut=--GET_CONNECTION--
msg=IsValid connOK=false released=false pooled=false tzOK=false
```

**Your oracle client version**
e.g. 12.2.0.1.0

**Your godror version**
e.g. v0.20.4

**Your go version**
e.g. 1.15.2

**Your gcc version**
e.g. 4.8.3

**Machine (please complete the following information):**
 - OS: [e.g. win]
 - Architecture: [e.g. x86_64]
 - Version: [e.g. 10]

**Additional context**

**Comment by tgulacsi** [comment]
PTAL 712fbc7 - I don't have a setup wher I can test it.

**Comment by kurt-google** [comment]
Sorry for the long turn around, but this does seem resolved in 0.25.2
